src/pb_stub.cc

```diff
@@ -837,6 +857,78 @@ Stub::GetOrCreateInstance()
   return Stub::stub_instance_;
 }
 
+void
+Stub::LaunchLogRequestThread()
+{
+  log_thread_ = true;
+  log_monitor_ = std::thread(&Stub::ServiceLogRequests, this);
+}
+
+void
+Stub::TerminateLogRequestThread()
+{
+  log_thread_ = false;
+  log_monitor_.join();
+}
+
+void
+Stub::EnqueueLogRequest(std::shared_ptr<PbLog> log_ptr)
+{
+  log_request_buffer.push(log_ptr);
```

Comment:
You need a synchronization logic to protect pushing messages to this queue. Otherwise, if multiple threads are pushing to this queue simultaneously there will be a race condition.